mangaki/mangaki/utils/mal.py

```diff
@@ -220,18 +229,72 @@ def search_works(self, work_type: MALWorks, query: str) -> List[MALEntry]:
                    getattr(settings, 'MAL_PASS', None))
 
 
-def insert_into_mangaki_database_from_mal(mal_entries: List[MALEntry]):
+def look_for_works(work_list: QuerySet,
+                   ext_poster: str,
+                   titles: List[str]) -> List[Work]:
+    constraints_work = None
+    constraints_title = None
+    for synonym in titles:
+        cons = SearchQuery(synonym, config='simple')
+
+        if constraints_work:
+            constraints_work |= cons
+        else:
+            constraints_work = cons
+
+        if constraints_title:
+            constraints_title |= cons
+        else:
+            constraints_title = cons
+
+    constraints_work = Q(title_search=constraints_work) | Q(ext_poster=ext_poster)
+
+    works = set(work_list.filter(constraints_work)[:2])
+    titles = list(
+        WorkTitle.objects
+            .filter(title_search=constraints_title)
+            .values_list('work_id', flat=True).all()
+    )
+    more_works = set(Work.objects.in_bulk(titles))
+
+    return list((works | more_works))
+
+
+def insert_into_mangaki_database_from_mal(mal_entries: List[MALEntry],
+                                          title: Optional[str] = None) -> Optional[Work]:
+    """
+    Insert into Mangaki database from MAL data.
+    And return the work corresponding to the title.
+    :param mal_entries: a list of MALEntry
+    :type mal_entries: List[MALEntry]
+    :param title: title of work looked for (optional)
+    :type title: str (can be None)
+    :return: a work if title is specified
+    :rtype: Optional[Work]
+    """
     # Assumption: Mangaki's slugs are the same as MAL's work types.
     if not mal_entries:
         return
 
+    first_work_matching = None
+
     # All entries are supposed of the same type.
     work_cat = Category.objects.get(slug=mal_entries[0].work_type.value)
     for entry in mal_entries:
-        is_present = Work.objects.filter(
-            category=work_cat,
-            ext_poster=entry.poster
-        ).exists()
+        # Check over all known titles.
+        titles = [entry.title]
+        if entry.english_title:
+            titles.append(entry.english_title)
+
+        is_present = (
```

Comment:
If there is a match, it may not be the work we are looking for, because MAL Search returns more results than there should be (works which are not relevant or not the ones we are looking for, but may be already in the database.)